azure-pipelines.yml

```diff
@@ -18,225 +20,44 @@ stages:
-          --build-arg GIT_COMMIT_HASH=$(Build.SourceVersion) \
-          .
-      displayName: Build web Docker image without cache
-      condition: and(eq(variables['Build.SourceBranch'], 'refs/heads/master'), succeeded())
-
-    # Push test images
-    - script: |
-        docker tag local/dfe-teachers-payment-service:test-dependencies $(dockerRegistry)/$(dockerImageName):cache-test-dependencies
-
-        docker push $(dockerRegistry)/$(dockerImageName):cache-test-dependencies
-      displayName: Push test dependencies Docker image for caching
-      condition: and(eq(variables['Build.SourceBranch'], 'refs/heads/master'), succeeded())
-
-    - script: |
-        docker tag local/dfe-teachers-payment-service:test $(dockerRegistry)/$(dockerImageName):cache-test
-
-        docker push $(dockerRegistry)/$(dockerImageName):cache-test
-      displayName: Push test Docker image for caching
-      condition: and(eq(variables['Build.SourceBranch'], 'refs/heads/master'), succeeded())
-
-    # Push web images
-    - script: |
-        docker tag local/dfe-teachers-payment-service:web-dependencies $(dockerRegistry)/$(dockerImageName):cache-web-dependencies
-
-        docker push $(dockerRegistry)/$(dockerImageName):cache-web-dependencies
-      displayName: Push web dependencies Docker image for caching
-      condition: and(eq(variables['Build.SourceBranch'], 'refs/heads/master'), succeeded())
-
-    - script: |
-        docker tag local/dfe-teachers-payment-service:web $(dockerRegistry)/$(dockerImageName):$(Build.BuildNumber)
-        docker tag local/dfe-teachers-payment-service:web $(dockerRegistry)/$(dockerImageName):latest
-
-        docker push $(dockerRegistry)/$(dockerImageName):$(Build.BuildNumber)
-        docker push $(dockerRegistry)/$(dockerImageName):latest
-      displayName: Push web Docker image
-      condition: and(eq(variables['Build.SourceBranch'], 'refs/heads/master'), succeeded())
-
-    - script: |
-        echo "Show Variables"
-        echo $(Build.BuildNumber)
-        echo $(dockerRegistry)
-        echo $(dockerImageName)
-
+    - template: templates/build.yml  #Run Build Template
 
 - stage: Release
   condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
   jobs:
-  - job: Terraform
-    steps:
-     - task: charleszipp.azure-pipelines-tasks-terraform.azure-pipelines-tasks-terraform-cli.TerraformCLI@0
-       displayName: 'Terraform Init'
-       inputs:
-         command: init
-         workingDirectory: $(Agent.BuildDirectory)/s/azure/terraform
-         backendType: azurerm
-         backendServiceArm: 'azdo.pipelines.cip.S118D.armfe1ef140-8bef-4043-b5ee-c449e6f951ef'
-         backendAzureRmResourceGroupName: 's118d01-tfbackend'
-         backendAzureRmStorageAccountName: s118d01tfbackendsa
-         backendAzureRmContainerName: s118d01conttfstate
-         backendAzureRmKey: terraform.tfstate
+  - job: DeployDev
+    variables:
+    - name: 'backendAzureRmContainerName'
```

Comment:
Have tried variable group at stage level like [here](https://github.com/SkillsFundingAgency/das-levy-transfer-matching-api/blob/76db6bed9b5228a42576abc71a6ddd267707ce1f/azure-pipelines.yml#L63 )  instead of defining single variables at job level?